plugins/editorconfig/src/org/editorconfig/language/codeinsight/inspections/EditorConfigDanglingDotInspection.kt

```diff
@@ -0,0 +1,27 @@
+// Copyright 2000-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+package org.editorconfig.language.codeinsight.inspections
+
+import com.intellij.codeInspection.LocalInspectionTool
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.psi.PsiElement
+import org.editorconfig.language.codeinsight.quickfixes.EditorConfigRemoveDanglingDotQuickFix
+import org.editorconfig.language.messages.EditorConfigBundle
+import org.editorconfig.language.psi.EditorConfigElementTypes
+import org.editorconfig.language.psi.EditorConfigQualifiedOptionKey
+import org.editorconfig.language.psi.EditorConfigVisitor
+
+class EditorConfigDanglingDotInspection : LocalInspectionTool() {
+  override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean) = object : EditorConfigVisitor() {
+    override fun visitQualifiedOptionKey(key: EditorConfigQualifiedOptionKey) {
+      assertNotDot(key.firstChild, key.firstChild.nextSibling)
```

Comment:
No, I did not define qualified key recursively as it would make it too complicated to match it againist a descriptor. Instead, it looks like
```
QualifiedName(foo.bar.baz)
_QualifiedKeyPart(foo)
_Dot
_QualifiedKeyPart(bar)
_Dot
_QualifiedKeyPart(baz)
```
So for a key `.foo.bar.baz` this inspection will fire